dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/guestbook/SelectGuestBookService.java

```diff
@@ -25,32 +24,54 @@ public SelectGuestBookService() {
     }
 
     @Inject
-    public SelectGuestBookService(DatasetVersionServiceBean versionService) {
-        this.versionService = versionService;
+    public SelectGuestBookService(DatasetService datasetService) {
+        this.datasetService = datasetService;
     }
 
     // -------------------- LOGIC --------------------
 
-    public Dataset saveGuestbookChanges(DatasetVersion editedDataset,
-                                        Option<Guestbook> selectedGuestbook,
-                                        Option<Guestbook> guestbookBeforeChanges) {
+    public Dataset saveGuestbookChanges(Dataset editedDataset,
+                                        Option<Guestbook> selectedGuestbook) {
 
-        Dataset dataset = editedDataset.getDataset();
+        Option<Guestbook> previousGuestbook = Option.of(editedDataset.getGuestbook());
 
-        if (isGuestbookAddedOrRemoved(selectedGuestbook, guestbookBeforeChanges)) {
+        if (guestbookIsAddedOrRemoved(selectedGuestbook, previousGuestbook)) {
+            editedDataset.setGuestbookChangeTime(Timestamp.from(Instant.now(systemTime)));
+        }
 
-            dataset.setGuestbookChangeTime(Timestamp.from(Instant.now(systemTime)));
+        if(isSelectedGuestbookSameAsPrevious(selectedGuestbook, previousGuestbook)) {
+            return editedDataset;
         }
 
-        dataset.setGuestbook(selectedGuestbook.getOrNull());
-        return versionService.updateDatasetVersion(editedDataset, true);
+        editedDataset.setGuestbook(selectedGuestbook.getOrNull());
+
+        return datasetService.updateDatasetGuestbook(editedDataset);
     }
 
     // -------------------- PRIVATE --------------------
 
-    private boolean isGuestbookAddedOrRemoved(Option<Guestbook> selectedGuestbook, Option<Guestbook> guestbookBeforeChanges) {
-        return (guestbookBeforeChanges.isEmpty() && selectedGuestbook.isDefined()) ||
-                (guestbookBeforeChanges.isDefined() && selectedGuestbook.isEmpty());
+    private boolean isSelectedGuestbookSameAsPrevious(Option<Guestbook> selectedGuestbook, Option<Guestbook> previousGuestbook) {
+        return isSameGuestbookChosen(selectedGuestbook, previousGuestbook) || isStillNoGuestbookChosen(selectedGuestbook, previousGuestbook);
+    }
+
+    private boolean isSameGuestbookChosen(Option<Guestbook> selectedGuestbook, Option<Guestbook> previousGuestbook) {
+        return selectedGuestbook.isDefined() && previousGuestbook.isDefined() &&  selectedGuestbook.get() == previousGuestbook.get();
```

Comment:
``` selectedGuestbook.get() == previousGuestbook.get()```

This is a bit risky on objects. Are we sure that we want to compare these only on the basis of reference?